### ISSUE TITLE
ngc in compiler-cli does not work after upgrade to angular 4

### ISSUE THREAD

**Issue by pearldiver** [angular/angular#16944]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
my project use "ngc -p ./tsconfig.lib.json" to compile an angular2 module.
After upgrade to angular 4, it prompt error:
Error at projectpath/src/lib/index.ts:7:42: Cannot find module '.folderfilename'.

Obviously, the file path is parsed incorrectly, there is no '/'. should be './folder/filename'
And after i go back to old compiler-cli (2.4.4), the compile works.

my tsconfig is:
{
  "compilerOptions": {
    "target": "es5",
    "module": "commonjs",
    "declaration": true,
    "stripInternal": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "noImplicitAny": false,
    "removeComments": true,
    "allowUnreachableCode": false,
    "moduleResolution": "node",
    "outDir": "./dist",
    "lib": ["es6", "dom"],
    "skipLibCheck": true,
    "types": []
  },
  "angularCompilerOptions": {
   "genDir": "./dist",
   "strictMetadataEmit": true,
   "skipTemplateCodegen": true,
   "annotateForClosureCompiler": true,
   "flatModuleOutFile": "index.js",
   "flatModuleId": "modulename",
   "debug": false
  },
  "files": [
    "./src/lib/index.ts"
  ],
  "compileOnSave": false,
  "buildOnSave": false
}

**Expected behavior**
ngc compile works

**Minimal reproduction of the problem with instructions**
it can reproduce by src with subfolders

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

Windows server

* **Angular version:** 2.0.X
"@angular/common": "^4.0.0",
    "@angular/core": "^4.1.3",
    "@angular/compiler": "^4.0.0",
    "@angular/compiler-cli": "^4.0.0",
    "@angular/forms": "^4.0.0",
    "@angular/http": "^4.0.0",
    "@angular/platform-browser": "^4.0.0",
    "@angular/platform-browser-dynamic": "^4.0.0",
    "@angular/router": "^4.0.0",

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web 
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
TypeScript 

* **Node (for AoT issues):** `node --version` =   
v6.9.5

**Comment by chuckjaz** [comment]
Thanks for reporting this issue. However, you didn't provide sufficient information for us to understand and reproduce the problem. Please check out [our submission guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-submitting-an-issue) to understand why we can't act on issues that are lacking important information.

If the problem still persists, please file a new issue and ensure you provide all of the required information when filling out the issue template.

It would be extremely helpful to have a repository that reproduces the issue that can be cloned.

**Comment by pearldiver** [comment]
sorry for the noise, I just found that the issue is caused by i installed the latest cli directly without clean up the node_modules.
what i did is
npm uninstall @angular/compiler-cli --save-dev
npm install @angular/compiler-cli@latest --save-dev
it has problems but after I deleted the whole node_modules and reinstall all packages, the ngc works. I don's exactly know which old module make the trouble.
since the problem is gone, close it.
